src/rebase.c

```diff
@@ -718,6 +719,11 @@ int git_rebase_init(
 		branch = head_branch;
 	}
 
+	if (!inmemory && upstream) {
+		if ((error = git_hook_call_pre_rebase(repo, upstream, (head_ref == NULL ? branch : NULL))) < 0)
```

Comment:
Do we really want to check for `< 0` here? Usually, return codes from exec'd programs are always positive in the range from 0 to 255, where non-zero positive values denote an error. So in case the callback simply exec's the hook and returns its value, we will not detect that error.

Reply:
I'm defaulting to the libgit2 convention. Right now the expectation is that the executor will bubble a `GIT_*` error, not the underlying exec return value.